support/modules/directory

```diff
@@ -26,16 +26,25 @@ case "$STATE" in
         test "$dest_dir" = "/" && \
             echo "Error: destination dir is '/', install not supported." && exit 1
         mkdir -p $dest_dir
-        if ! tar -cf ${prev_files_tar} -C ${dest_dir} .
+        
+        tar -cf ${prev_files_tar} -C ${dest_dir} .
+        createPrevSuccess=$?
```

Comment:
I think this won't work as intended because `set -e` is active. It will simply stop at the `tar` command once there is an error, because it's not part of a conditional, and won't continue to the next line.

I can't really see the problem with the previous code though, it looks correct to me. `ret` is being set, and is returned, as it should. The only thing missing is a more descriptive message. Are you saying this didn't work correctly for you?